src/core/types/Api.ts

```diff
@@ -0,0 +1,49 @@
+import { Address, Hex } from './Common';
+
+export type ApiResponse<T = unknown> = {
+  success: boolean;
+  data?: T;
+  error?: {
+    type?: string;
+    message: string;
+    code?: string;
+    details?: unknown;
```

Comment:
what's the usage of this `unknown` type of `details`?

Reply:
Not sure will remove